### PR TITLE
Unify numeric types conversion utilities. Introduce Numbers.

### PR DESCRIPTION
**Pull request by MishaDemianenko**
Remove duplicated conversions across code base that perform
safe conversion from one numeric type to another. Unify them and cleanup.

### CODE REVIEW

File: community/common/src/main/java/org/neo4j/helpers/Numbers.java [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+    public static short safeCastIntToUnsignedShort( int value )
+    {
+        if ( (value & ~0xFFFF) != 0 )
+        {
+            throw new ArithmeticException( getOverflowMessage( value, "unsigned short" ) );
+        }
+        return (short) value;
+    }
+
+    public static int safeCastLongToInt( long value )
```
Comment (tinwelint) [inline community/common/src/main/java/org/neo4j/helpers/Numbers.java:33]: Isn't this `Math.toIntExact` btw?

Comment (MishaDemianenko) [reply community/common/src/main/java/org/neo4j/helpers/Numbers.java:33]: Almost, except error message that aligns with other safe casts error messages

File: community/common/src/test/java/org/neo4j/helpers/NumbersTest.java [hunk 2]
```diff
@@ -0,0 +1,120 @@
...
+        assertEquals(Short.MAX_VALUE, safeCastLongToShort( Short.MAX_VALUE ));
+        assertEquals(Short.MIN_VALUE, safeCastLongToShort( Short.MIN_VALUE ));
+    }
+
+    @Test
+    public void castIntToUnsighedShort()
+    {
+        assertEquals(1, safeCastIntToUnsignedShort( 1 ));
+        assertEquals(10, safeCastIntToUnsignedShort( 10 ));
+        assertEquals(Short.MAX_VALUE, safeCastIntToUnsignedShort( Short.MAX_VALUE ));
```
Comment (burqen) [inline community/common/src/test/java/org/neo4j/helpers/NumbersTest.java]: We are not testing the largest value for an unsigned short here.

**Review by burqen — CHANGES_REQUESTED** [review]
Found a missing test case, otherwise this looks fine 🎉 
Nice cleanup!

**Comment by MishaDemianenko** [comment]
@burqen updated

**Review by burqen — APPROVED** [review]

**Pull request merged**